packages/react/src/components/Tile/Tile.js

```diff
@@ -181,212 +181,207 @@ export class ClickableTile extends Component {
-    ),
-
-    /**
-     * The ID of the `<input>`.
-     */
-    id: PropTypes.string,
-
-    /**
-     * `true` to use the light version. For use on $ui-01 backgrounds only.
-     * Don't use this to make tile background color same as container background color.
-     */
-    light: PropTypes.bool,
-
-    /**
-     * The `name` of the `<input>`.
-     */
-    name: PropTypes.string,
-
-    /**
-     * The empty handler of the `<input>`.
-     */
-    onChange: PropTypes.func,
-
-    /**
-     * `true` to select this tile.
-     */
-    selected: PropTypes.bool,
-
-    /**
-     * Specify the tab index of the wrapper element
-     */
-    tabIndex: PropTypes.number,
-
-    /**
-     * The `title` of the `<input>`.
-     */
-    title: PropTypes.string,
-
-    /**
-     * The value of the `<input>`.
-     */
-    value: PropTypes.oneOfType([PropTypes.string, PropTypes.number]).isRequired,
-  };
-
-  static defaultProps = {
-    value: 'value',
-    title: 'title',
-    selected: false,
-    handleClick: () => {},
-    handleKeyDown: () => {},
-    onChange: () => {},
-    tabIndex: 0,
-    light: false,
-  };
-
-  static getDerivedStateFromProps({ selected }, state) {
-    const { prevSelected } = state;
-    return prevSelected === selected
-      ? null
-      : {
-          selected,
-          prevSelected: selected,
-        };
-  }
-
-  handleClick = (evt) => {
+export function SelectableTile(props) {
+  const {
+    children,
+    id,
+    tabIndex = 0,
+    value,
+    name,
+    title,
+    // eslint-disable-next-line no-unused-vars
+    iconDescription,
+    className,
+    handleClick,
+    handleKeyDown,
+    onClick = () => {},
+    onChange = () => {},
+    onKeyDown = () => {},
+    light,
+    disabled,
+    selected,
+    ...other
+  } = props;
+
+  // TODO: replace with onClick when handleClick prop is deprecated
+  const clickHandler = handleClick || onClick;
+
+  // TODO: replace with onClick when handleClick prop is deprecated
+  const keyDownHandler = handleKeyDown || onKeyDown;
+
+  const [isSelected, setIsSelected] = useState(selected);
+  const input = useRef(null);
+  const classes = classNames(
+    `${prefix}--tile`,
+    `${prefix}--tile--selectable`,
+    {
+      [`${prefix}--tile--is-selected`]: isSelected,
+      [`${prefix}--tile--light`]: light,
+      [`${prefix}--tile--disabled`]: disabled,
+    },
+    className
+  );
+
+  // TODO: rename to handleClick when handleClick prop is deprecated
+  const handleOnClick = (evt) => {
     evt.preventDefault();
     evt.persist();
-    this.setState(
-      {
-        selected: !this.state.selected,
-      },
-      () => {
-        this.props.handleClick(evt);
-        this.props.onChange(evt);
-      }
-    );
+    setIsSelected(!isSelected);
+    clickHandler(evt);
+    onChange(evt);
   };
 
-  handleKeyDown = (evt) => {
+  // TODO: rename to handleKeyDown when handleKeyDown prop is deprecated
+  const handleOnKeyDown = (evt) => {
```

Comment:
One quick style nit: for refactors we should make sure we're following the style guide for components with hooks: [Component](https://github.com/carbon-design-system/carbon/blob/main/docs/style.md#writing-a-component) and [Event Handlers](https://github.com/carbon-design-system/carbon/blob/main/docs/style.md#naming-event-handlers)